src/main/scala/edu/ie3/simona/model/em/OptimizedFlexStrat.scala

```diff
@@ -0,0 +1,338 @@
+/*
+ * © 2025. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.em
+
+import edu.ie3.datamodel.models.input.AssetInput
+import edu.ie3.simona.exceptions.CriticalFailureException
+import edu.ie3.simona.model.em.OptimizedFlexStrat.*
+import edu.ie3.simona.ontology.messages.flex.MathFlexOptions
+import edu.ie3.simona.ontology.messages.flex.MathFlexOptions.{
+  OperationVars,
+  SoftConstraint,
+}
+import optimus.algebra.{Double2Const, Expression, Zero}
+import optimus.optimization.MPModel
+import optimus.optimization.enums.{SolutionStatus, SolverLib}
+import optimus.optimization.model.MPFloatVar
+import org.slf4j.Logger
+import squants.{Power, Time}
+
+import java.util.UUID
+
+/** Flex strategy that optimizes over a fixed amount of time steps into the
+  * future. Works with [[MathFlexOptions]], which describe constraints on
+  * present and future behavior of a participant.
+  *
+  * @param sampleTime
+  *   The amount of time in between steps.
+  * @param predictionHorizon
+  *   The amount of time that is considered in the future, i.e. the last step is
```

Comment:
```suggestion
  *   The amount of time that is predicted into the future, i.e. the last step is
```

as i would interpret a "predictionHorizon"